presto-product-tests/src/main/java/io/prestosql/tests/kafka/TestKafkaAvroWritesSmokeTest.java

```diff
@@ -0,0 +1,153 @@
+    @Requires(AllDataTypesAvroTable.class)
+    public void testInsertPrimitiveDataType()
+    {
+        assertThat(query(format(
+                "insert into %s.%s values " +
+                        "('foobar', 127, 234.567, true)",
+                KAFKA_CATALOG,
+                ALL_DATATYPES_AVRO_TABLE_NAME
+        ))).updatedRowsCountIsEqualTo(1);
+
+        assertThat(query(format(
+                "select * from %s.%s",
+                KAFKA_CATALOG,
+                ALL_DATATYPES_AVRO_TABLE_NAME
+        ))).containsOnly(
+                row("foobar", 127, 234.567, true));
+    }
+
+    private static class NullDataAvroTable
+            implements RequirementsProvider
+    {
+        @Override
+        public Requirement getRequirements(Configuration configuration)
+        {
+            return immutableTable(new KafkaTableDefinition(
+                    ALL_NULL_AVRO_TABLE_NAME,
+                    ALL_NULL_AVRO_TOPIC_NAME,
+                    new ListKafkaDataSource(ImmutableList.of()),
+                    1,
+                    1));
+        }
+    }
+
+    @Test(groups = {KAFKA, PROFILE_SPECIFIC_TESTS})
+    @Requires(NullDataAvroTable.class)
+    public void testNullType()
+    {
+        assertThat(query(format(
+                "insert into %s.%s values " +
+                        "(null, null, null, null)",
+                KAFKA_CATALOG,
+                ALL_NULL_AVRO_TABLE_NAME
+        ))).updatedRowsCountIsEqualTo(1);
+
+        assertThat(query(format(
+                "select * from %s.%s",
+                KAFKA_CATALOG,
+                ALL_NULL_AVRO_TABLE_NAME
+        ))).containsOnly(
+                row(null, null, null, null));
+    }
+
+    private static class StructuralDataTypeTable
+            implements RequirementsProvider
+    {
+        @Override
+        public Requirement getRequirements(Configuration configuration)
+        {
+            return immutableTable(new KafkaTableDefinition(
+                    STRUCTURAL_AVRO_TABLE_NAME,
+                    STRUCTURAL_AVRO_TOPIC_NAME,
+                    new ListKafkaDataSource(ImmutableList.of()),
+                    1,
+                    1));
+        }
+    }
+
+    @Test(groups = {KAFKA, PROFILE_SPECIFIC_TESTS})
+    @Requires(StructuralDataTypeTable.class)
+    public void testInsertStructuralDataType()
+    {
+        // TODO Add support for structural types in AvroRowEncoder
+        try {
+            query(format(
+                    "insert into %s.%s values " +
+                            "(ARRAY[100, 102], map_from_entries(ARRAY[('key1', 'value1')]))",
+                    KAFKA_CATALOG,
+                    STRUCTURAL_AVRO_TABLE_NAME));
+        }
+        catch (RuntimeException e) {
+            // java.lang.IllegalArgumentException: Unsupported column type 'array(bigint)' for column 'c_array'
+            assertTrue(nullToEmpty(e.getMessage()).matches(".*Unsupported column type.*"));
```

Comment:
This is too weak, since the test passes _also when_ the query doesn't throw.

Use dedicated tempto method for that, see
https://github.com/prestosql/presto/blob/2af0ee3cfe1ca9f0dae26dc60d1d284949e4dc4f/presto-product-tests/src/main/java/io/prestosql/tests/TestAlterTable.java#L70-L71

Also, instead of putting expected message in a comment, just include it in the `.matches` in its full beauty